### PR TITLE
Update build to fix CI

### PR DESCRIPTION
**Pull request by agoscinski**
Mostly just updating all the files and dependencies.

Maybe controversial change is that I removed the `-W` flag during the sphinx-build process, so warnings are treated as warnings and not as errors. The reason for this is because I got sphinx warnings which in my experience are very hard to debug and only marginal add to the quality to the doc page. I will not fix these warnings.

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -56,7 +56,7 @@ cleannotebooks:
 
 # -W: exit on warning
 defaultdoc:
-	$(SPHINXBUILD) -b html -nW $(ALLSPHINXOPTS) $(BUILDDIR)/html
+	$(SPHINXBUILD) -b html -n $(ALLSPHINXOPTS) $(BUILDDIR)/html
```
Comment (GeigerJ2) [inline docs/Makefile]: This disables `fail_on_warning`, though in the `.readthedocs.yml` file it's still set to true?

File: requirements.txt
Comment (GeigerJ2) [inline requirements.txt:1]: Shouldn't we pin the versions here?

File: docs/conf.py [hunk 2]
```diff
@@ -390,6 +390,9 @@
 
 suppress_warnings = ["misc.highlighting_failure"]
 
+# Links we ignore, because they do not work temporary and we cannot fix it
+linkcheck_ignore = ["https://www.virtualbox.org/", "https://www.big-map.eu/"]
```
Comment (GeigerJ2) [inline docs/conf.py]: www.virtualbox.org works for me.

**Review by GeigerJ2 — APPROVED** [review]
Not really an expert on CI stuff, but changes look reasonable to me. If this fixes the broken CI, and we can merge the other PRs, I'd say we can go for it.

File: docs/conf.py (hunk 2 above)
Comment (agoscinski) [reply docs/conf.py]: Yes it hat some down yesterday, not sure what to do. Maybe I just was unlucky, I can remove it again.

Comment (agoscinski) [reply docs/conf.py]:
```suggestion
linkcheck_ignore = ["https://www.big-map.eu/"]
```

File: docs/Makefile (hunk 1 above)
Comment (agoscinski) [reply docs/Makefile]: True,  but the sphinx-build on RTD also has the `--keep-going` which overwrites the `-W`. A bit weird to me, but in the end it is the same
https://readthedocs.org/projects/aiida-tutorials/builds/25575956/
I just use the same flags as on RTD
```suggestion
	$(SPHINXBUILD) -b html -T -W --keep-going $(ALLSPHINXOPTS) $(BUILDDIR)/html
```

File: requirements.txt [hunk 3]
```diff
@@ -1,7 +1,7 @@
-jinja2~=3.0
-Sphinx~=4.1
+jinja2
+Sphinx
```
Comment (agoscinski) [inline requirements.txt]:
```suggestion
jinja2~=3.1.4
Sphinx~=8.0.2
```

File: requirements.txt [hunk 4]
```diff
@@ -1,7 +1,7 @@
...
 sphinx-book-theme
```
Comment (agoscinski) [inline requirements.txt]:
```suggestion
sphinx-book-theme~=1.1.3
```

File: requirements.txt [hunk 5]
```diff
@@ -1,7 +1,7 @@
...
-sphinx-copybutton~=0.4.0
+sphinx-copybutton
 sphinx-panels
 sphinx-tabs
```
Comment (agoscinski) [inline requirements.txt]:
```suggestion
sphinx-copybutton~=0.5.2
sphinx-panels~=0.4.1
sphinx-tabs~=3.4.5
```

File: requirements.txt [hunk 6]
```diff
@@ -1,7 +1,7 @@
...
-myst-nb==0.16.0
+myst-nb
```
Comment (agoscinski) [inline requirements.txt]:
```suggestion
myst-nb~=1.1.1
```

File: requirements.txt
Comment (agoscinski) [reply requirements.txt:1]: We can pin them. Sometimes it is more annoying than helpful to pin verison if a repo changes frequently and needs to update its dependencies. But I guess in this case we only need to update the occasionally.

Comment (GeigerJ2) [reply requirements.txt:1]: Yeah, I usually had the same experience so far, and ended up not pinning them usually :D

File: docs/Makefile (hunk 1 above)
Comment (GeigerJ2) [reply docs/Makefile]: Yeah, a bit weird, I agree.

**Pull request merged**